### PR TITLE
Improve makefiles for documentation

### PR DESCRIPTION
**Pull request by mppf**
Add new top-level makefile targets that are easier for
me to type correctly (I keep forgetting one or both s's
in make modules-docs) and also docs as an easy target.

Add clean-documentation target and arrage for make clobber
to call it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,6 +100,10 @@ modules-docs: chpldoc
 # dependency so parallel make executions correctly build chpldoc first.
 	$(MAKE) modules-docs-only
 
+module-docs: modules-docs
```
Comment (lydia-duncan) [inline Makefile]: I'd be fine with replacing modules-docs with module-docs myself, but that's not my call.

Comment (bradcray) [inline Makefile]: I'd like to replace it too.  I kept typing the wrong thing every time I tried this.  Ditto modules-docs-only => module-docs-only.

**Comment by thomasvandoren** [comment]
:+1:

**Comment by mppf** [comment]
OK, I've updated it to remove modules-docs and only have module-docs (per brad/lydia).

**Comment by lydia-duncan** [comment]
Awesome, thanks!

**Pull request merged**